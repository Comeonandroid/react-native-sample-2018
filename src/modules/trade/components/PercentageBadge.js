import React, { PureComponent } from 'react'
import PropTypes from 'prop-types'
import styled, { css } from 'styled-components'
import { ifProp, prop } from 'styled-tools'

import { View, Text } from '../../../elements'

class PercentageBadge extends PureComponent {
  render() {
    const { upraising, children } = this.props
    return (
      <Wrapper upraising={upraising}>
        <Element upraising={upraising}>{children}</Element>
      </Wrapper>
    )
  }
}

PercentageBadge.propTypes = {
  children: PropTypes.node,
  upraising: PropTypes.bool
}

const Wrapper = styled(View) `
  margin-left: 10px;
  padding: 4px 20px;
  borderRadius: 2px;
  ${ifProp('upraising',
    css`
    background: ${ prop('theme.colors.success')};
  `,
    css`
    background: ${ prop('theme.colors.danger')};
  `)}
`
const Element = styled(Text) `
  font-family: ${prop('theme.typo.condensedFont')};
  font-size: 16px;
  font-weight: 400;
  color: ${ifProp({ upraising: true }, prop('theme.colors.buttonLight'), prop('theme.colors.buttonDark'))}
`

export default PercentageBadge
