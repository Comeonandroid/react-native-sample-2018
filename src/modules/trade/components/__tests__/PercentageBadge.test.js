import 'react-native'
import React from 'react'

import { itBehavesLike } from '../../../../specs/shared_examples/index'

import PercentageBadge from '../PercentageBadge'

const setup = propOverrides => {
  const props = Object.assign({
    children: '50',
    upraising: false
  }, propOverrides)

  const component = <PercentageBadge {...props} />

  return {
    props,
    component
  }
}

describe('<PercentageBadge />', () => {
  const { component } = setup()
  itBehavesLike('aThemedComponent', { subject: component })

  describe('upraising: true', () => {
    const { component } = setup({ active: 'value 2' })
    itBehavesLike('aThemedComponent', { subject: component })
  })
})