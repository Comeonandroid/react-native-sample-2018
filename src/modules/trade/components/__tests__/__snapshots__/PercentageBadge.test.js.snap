// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PercentageBadge /> dark: renders correctly 1`] = `
<View
  style={
    Array [
      Object {},
      Array [
        Object {
          "backgroundColor": "#f25c74",
          "borderRadius": 2,
          "marginLeft": 10,
          "paddingBottom": 4,
          "paddingLeft": 20,
          "paddingRight": 20,
          "paddingTop": 4,
        },
        undefined,
      ],
    ]
  }
  upraising={false}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#fcfcfc",
          "fontFamily": "Open Sans",
        },
        Array [
          Object {
            "color": "#000",
            "fontFamily": "Roboto Condensed",
            "fontSize": 16,
            "fontWeight": "400",
          },
          undefined,
        ],
      ]
    }
    upraising={false}
  >
    50
  </Text>
</View>
`;

exports[`<PercentageBadge /> light: renders correctly 1`] = `
<View
  style={
    Array [
      Object {},
      Array [
        Object {
          "backgroundColor": "#f25c74",
          "borderRadius": 2,
          "marginLeft": 10,
          "paddingBottom": 4,
          "paddingLeft": 20,
          "paddingRight": 20,
          "paddingTop": 4,
        },
        undefined,
      ],
    ]
  }
  upraising={false}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#000000",
          "fontFamily": "Open Sans",
        },
        Array [
          Object {
            "color": "#000",
            "fontFamily": "Roboto Condensed",
            "fontSize": 16,
            "fontWeight": "400",
          },
          undefined,
        ],
      ]
    }
    upraising={false}
  >
    50
  </Text>
</View>
`;

exports[`<PercentageBadge /> upraising: true dark: renders correctly 1`] = `
<View
  style={
    Array [
      Object {},
      Array [
        Object {
          "backgroundColor": "#f25c74",
          "borderRadius": 2,
          "marginLeft": 10,
          "paddingBottom": 4,
          "paddingLeft": 20,
          "paddingRight": 20,
          "paddingTop": 4,
        },
        undefined,
      ],
    ]
  }
  upraising={false}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#fcfcfc",
          "fontFamily": "Open Sans",
        },
        Array [
          Object {
            "color": "#000",
            "fontFamily": "Roboto Condensed",
            "fontSize": 16,
            "fontWeight": "400",
          },
          undefined,
        ],
      ]
    }
    upraising={false}
  >
    50
  </Text>
</View>
`;

exports[`<PercentageBadge /> upraising: true light: renders correctly 1`] = `
<View
  style={
    Array [
      Object {},
      Array [
        Object {
          "backgroundColor": "#f25c74",
          "borderRadius": 2,
          "marginLeft": 10,
          "paddingBottom": 4,
          "paddingLeft": 20,
          "paddingRight": 20,
          "paddingTop": 4,
        },
        undefined,
      ],
    ]
  }
  upraising={false}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#000000",
          "fontFamily": "Open Sans",
        },
        Array [
          Object {
            "color": "#000",
            "fontFamily": "Roboto Condensed",
            "fontSize": 16,
            "fontWeight": "400",
          },
          undefined,
        ],
      ]
    }
    upraising={false}
  >
    50
  </Text>
</View>
`;
